Marlin/Marlin_main.cpp

```diff
@@ -8276,8 +8271,8 @@ void ok_to_send() {
       SERIAL_ECHOPAIR(" L=", L);
       SERIAL_ECHOPAIR(" R=", R);
       SERIAL_ECHOLNPAIR(" offset=", offset);
+      last_offset = offset;
```

Comment:
The test is meant to deal with sudden changes from one segment to another, not differences from start to destination. It is expected that the end of a long line might be at a much different height from the start. The intent here was, and is, to flag any anomalous sudden change in Z across a small distance.